buildpack/telemetry/dynatrace.py

```diff
@@ -5,23 +5,148 @@
+    # "DT_PAAS_TOKEN": required, also used for telegraf integration
+    # "DT_SAAS_URL": required, also used for telegraf integration
+    "DT_TENANT": None,  # required for agent integration, dynatrace environment ID
+    "DT_TENANTTOKEN": None,  # optional, default value is get from manifest.json which is downloaded along with the agent installer
+    # -- Environment variables for orchestration
+    "DT_LOCALTOVIRTUALHOSTNAME": None,  # optional, default not set
+    "DT_APPLICATIONID": None,  # optional, default not set
+    "DT_NODE_ID": None,  # optional, default not set
+    "DT_CLUSTER_ID": None,  # optional, default not set
+    "DT_TAGS": None,  # optional tags e.g. MikesStuff easyTravel=Mike
+    "DT_CUSTOM_PROP": None,  # optional metadata e.g. Department=Acceptance Stage=Sprint
+    # -- Environment variables for troubleshooting
+    "DT_LOGSTREAM": "stdout",  # optional
+    "DT_LOGLEVELCON": None,  # Use this environment variable to define the console log level. Valid options are: NONE, SEVERE, and INFO.
+    "DT_AGENTACTIVE": None,  # Set to true or false to enable or disable OneAgent.
+    # -- Networking environment variables
+    "DT_NETWORK_ZONE": None,  # optional, Specifies to use a network zone. For more information
+    "DT_PROXY": None, # Optional, When using a proxy, use this environment variable to pass proxy credentials.
+}
+
+
+def stage(buildpack_dir, build_path, cache_path):
+    """
+    Downloads and unzips necessary OneAgent components
+    """
+    if is_agent_enabled():
+        try:
+            util.resolve_dependency(
+                "dynatrace.agent",
+                build_path,  # DOT_LOCAL_LOCATION,
+                buildpack_dir=buildpack_dir,
+                cache_dir=cache_path,  # CACHE_DIR,
+                unpack=True,
+                overrides={
+                    "url": os.environ.get("DT_SAAS_URL"),
+                    "environment": os.environ.get("DT_TENANT"),
+                    "token": os.environ.get("DT_PAAS_TOKEN"),
+                },
+                ignore_cache=True
+            )
+        except Exception as e:
+            logging.warning(
+                "Dynatrace agent download and unpack failed", exc_info=True
+            )
+
+
+def update_config(m2ee, app_name):
+    """
+    Injects Dynatrace configuration to java runtime
+    """
+    if not is_agent_enabled():
+        logging.debug(
+            "Skipping Dynatrace OneAgent setup, required env vars are not set"
+        )
+        return
+    logging.info("Enabling Dynatrace OneAgent")
+    try:
+        manifest = get_manifest()
+    except Exception as e:
+        logging.warning(
+            "Failed to parse Dynatrace manifest file", exc_info=True
+        )
+        return
+
+    # dynamic default
+    default_env.update({"DT_TENANTTOKEN": manifest.get("tenantToken")})
+
+    for key, dv in default_env.items():
+        value = os.environ.get(key, dv)
+        if value:
```

Comment:
I think we cannot do anything about empty string. If sb sets an environment variable to an empty string explicitly it is a valid input I'd say. 
Boolean values are more ambiguous to me in this case because we get these values from environment variables and inject them into the running java process. I haven't looked for how boolean values can be transferred without being converted to a string but in our case, we already don't have any boolean-valued environment variables.
So, treating any string input from the environment variables as valid seems ok to me in this case.

Reply:
Exactly, but this check won't allow empty strings or False values.

So the check should be :
```if value is not None:```

Reply:
Yeah, that's true. updated